staging/src/k8s.io/kms/service/service.go

```diff
@@ -0,0 +1,116 @@
+package service
+
+import (
+	"context"
+	"encoding/base64"
+
+	"k8s.io/klog/v2"
+	api "k8s.io/kms/apis/v2alpha1"
+	"k8s.io/kms/encryption"
+)
+
+const (
+	version              = "v2alpha1"
+	encryptedLocalKEKKey = "encryptedLocalKEK"
+)
+
+// Service offers encryption and decryption cache upfront of an remote KMS.
+type Service struct {
+	managedKeys *encryption.ManagedCipher
+}
+
+var (
+	_ api.KeyManagementServiceServer = (*Service)(nil)
+)
+
+// NewKeyManagementService creates an v2alpha1.KeyManagementServiceServer that
+// can be used for encryption and decryption, if given an remote encryption
+// service (remote KMS).
+func NewKeyManagementService(remoteCipher encryption.EncrypterDecrypter) (api.KeyManagementServiceServer, error) {
```

Comment:
The kms provider implementer would invoke `NewKeyManagementService` in the provider to initialize? If that's the case shouldn't we allow the implementer to configure the cache size with an options struct?